src/MSBuild/XMake.cs

```diff
@@ -832,6 +836,10 @@ private static void DebuggerLaunchCheck()
                     // if there was no need to start the build e.g. because /help was triggered
                     // do nothing
                 }
+
+                // The encoding may be changed to support non-en characters for environment variables set by external tools. We don't want to impact other programs on the console.
+                Console.OutputEncoding = originalOutputEncoding;
+                Console.InputEncoding = originalInputEncoding;
```

Comment:
I don't understand this. Shouldn't our program exiting reset this to whatever was present before?

Reply:
That does not happen.
You can see in the SDK: 
![image](https://user-images.githubusercontent.com/23152278/223586003-0bc01908-535a-45f1-b4c9-d74b8bccd56f.png)
If the code to change it back at the end is disabled, even upon exit running chcp shows the code page is changed. There was also a customer who reported this as an issue (so not just a dogfood thing.) Unfortunately, there is no doc for this.